File: src/main/java/com/project/vrs/mongo/settings/model/Notification.java
package com.project.vrs.mongo.settings.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.vrs.enums.NotificationStatus;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.Date;
import java.util.Map;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Notification implements Serializable {

    private Long id;

    private String title;

    private String message;

    @JsonProperty("notification_id")
    private String notificationId;

    @JsonProperty("user_id")
    private String userId;

    @JsonProperty("driver_id")
    private String driverId;

    private Map<String, String> properties;

    @JsonProperty("notification_status")
    private NotificationStatus notificationStatus;

    @JsonProperty("notification_read_status")
    private boolean notificationReadStatus;

    private boolean secured;

    @JsonProperty("notification_count")
    private long notificationCount;

    @JsonProperty("created_at")
    private Date createdAt;
}

File: client/src/app/model/notification.ts
// Define the enum for NotificationStatus
enum NotificationStatus {
  PENDING = "PENDING",
  SENT = "SENT",
  READ = "READ",
  FAILED = "FAILED"
}

// Define the interface for Notification
interface Notification {
  id: number;
  // @ts-ignore
  title: string;
  message: string;
  notification_id: string;
  user_id: string;
  properties: { [key: string]: string };
  notification_status: NotificationStatus;
  notification_read_status: boolean;
  secured: boolean;
  notification_count: number;
}
